analyzers/src/SonarAnalyzer.Utilities/RuleFinder.cs

```diff
@@ -39,46 +39,37 @@ public class RuleFinder
                 Assembly.Load(typeof(Rules.Common.DoNotInstantiateSharedClassesBase).Assembly.GetName())
             };
 
-        internal IEnumerable<Type> AllAnalyzerTypes { get; } // Rule-only, without utility analyzers
+        internal IEnumerable<Type> RuleAnalyzerTypes { get; } // Rule-only, without utility analyzers
         internal IEnumerable<Type> UtilityAnalyzerTypes { get; }
 
         public RuleFinder()
         {
             var all = PackagedRuleAssemblies
                 .SelectMany(assembly => assembly.GetTypes())
-                .Where(x => x.IsSubclassOf(typeof(DiagnosticAnalyzer)))
+                .Where(x => x.IsSubclassOf(typeof(DiagnosticAnalyzer)) && x.GetCustomAttributes<DiagnosticAnalyzerAttribute>().Any())
                 .ToArray();
 
-            AllAnalyzerTypes = all.Where(x => x.GetCustomAttributes<RuleAttribute>().Any()).ToList();
-            UtilityAnalyzerTypes = all.Where(x => typeof(UtilityAnalyzerBase).IsAssignableFrom(x) && x.GetCustomAttributes<DiagnosticAnalyzerAttribute>().Any()).ToList();
+            RuleAnalyzerTypes = all.Where(x => !typeof(UtilityAnalyzerBase).IsAssignableFrom(x)).ToList();
+            UtilityAnalyzerTypes = all.Where(x => typeof(UtilityAnalyzerBase).IsAssignableFrom(x)).ToList();
```

Comment:
```suggestion
            UtilityAnalyzerTypes = all.Where(x => typeof(UtilityAnalyzerBase).IsAssignableFrom(x)).ToList();
            RuleAnalyzerTypes = all.Except(UtilityAnalyzerTypes).ToList();
```

Reply:
I did a quick perf test and the `Except` is 3 times slower than the wordy version.